tox.ini

```
@@ -179,3 +179,30 @@ install_command = pip install --no-deps {opts} {packages}
 commands =
     conda list
     {env:PYTEST_COMMAND} {posargs}
+
+[testenv:pyinstaller]
+description = check that sunpy can be included in a pyinstaller bundle
+changedir = .pyinstaller
+deps =
+    # The developer version of PyInstaller doesn't work yet because it
+    # requires https://github.com/pyinstaller/pyinstaller/pull/4888 -
+    # so for now we pin to the latest working commit.
+    git+https://github.com/pyinstaller/pyinstaller.git@c111ec5c#egg=PyInstaller
+    # PyInstaller is not yet compatible with Matplotlib 3.3 so we pin
+    # Matplotlib to an older version for now - see the following issue
+    # for more details: https://github.com/pyinstaller/pyinstaller/issues/5004
+    matplotlib<3.3
+extras =
+    tests
+    net
+    all
+commands =
+    pyinstaller --onefile run_sunpy_tests.py \
+                --distpath . \
```

If my suggestion about the test command works we wouldn't need this?
```suggestion
    pyinstaller --distpath . \
```

---

Doing this would create an entire directory of files, as I mentioned earlier we want to test the executable for SunPy. We're making an `exe` and then running it.